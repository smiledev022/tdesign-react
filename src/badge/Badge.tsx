import React from 'react';
import classNames from 'classnames';
import { StyledProps } from '../_type';
import useConfig from '../_util/useConfig';

export interface BadgeProps extends StyledProps {
  /**
   * 自定义颜色
   */
  color?: string;
  /**
   * 展示的数字
   */
  count?: number;
  /**
   * 是否为红点
   * @default false
   */
  dot?: boolean;
  /**
   * 封顶数字值
   * @default 99
   */
  maxCount?: number;
  /**
   * 自定义文字，优先于 count
   */
  content?: React.ReactNode;
  /**
   * 尺寸
   * @default medium
   */
  size?: 'medium' | 'small';
  /**
   * 形状
   * @default circle
   */
  shape?: 'circle' | 'round';
  /**
   * 数值为 0 时是否展示
   * @default false
   */
  showZero?: boolean;
  /**
   * 设置状态点的位置偏移，格式 [x, y]
   */
  offset?: [number, number];
}

const Badge: React.FC<BadgeProps> = ({
  color,
  count = null,
  dot = false,
  maxCount = 99,
  content,
  size = 'medium',
  shape = 'circle',
  showZero = false,
  offset = [],
  className,
  children = null,
  style = {},
  ...restProps
}) => {
  const { classPrefix } = useConfig();

  const badgeClassName = classNames(
    !children && `${classPrefix}-badge-static`,
    dot ? `${classPrefix}-badge--dot` : shape && `${classPrefix}-badge--${shape}`,
    size === 'small' && `${classPrefix}-size-s`,
    !children && className,
  );
  const getDisplayCount = () => ((count as number) > maxCount ? `${maxCount}+` : count);

  const isHidden = !content && (typeof count !== 'number' || (count < 1 && !showZero));

  const getStyle = () => {
    const mergedStyle: React.CSSProperties = { ...style };
    if (color) mergedStyle.backgroundColor = color;
    if (offset) {
      if (offset[0]) {
        mergedStyle.right = -offset[0];
      }
      if (offset[1]) {
        mergedStyle.marginTop = -offset[1];
      }
    }
    return mergedStyle;
  };

  const badge = !isHidden ? (
    <span {...(children ? {} : restProps)} className={badgeClassName} style={getStyle()}>
      {!dot ? content || getDisplayCount() : null}
    </span>
  ) : null;

  if (!children) return badge;

  return (
    <span {...restProps} className={classNames(`${classPrefix}-badge`, className)}>
      {children}
      {badge}
    </span>
  );
};

Badge.displayName = 'Badge';

export default Badge;
