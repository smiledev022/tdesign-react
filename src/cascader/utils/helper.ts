import { TreeNode } from '../interface';

/**
 * 获取node节点的全路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}
