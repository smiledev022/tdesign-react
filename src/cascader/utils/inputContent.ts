import { TreeNode, CascaderContextType, TreeNodeValue } from '../interface';

/**
 * icon Class
 * @param prefix
 * @param CLASSNAMES
 * @param cascaderContext
 * @returns
 */
export function getIconClass(prefix: string, CLASSNAMES: any, cascaderContext: CascaderContextType) {
  const { visible } = cascaderContext;
  return [
    `${prefix}-cascader-icon`,
    {
      [CLASSNAMES.STATUS.visible]: visible,
    },
  ];
}

/**
 * Cascader外显div class
 * @param prefix
 * @param CLASSNAMES
 * @param cascaderContext
 * @returns
 */
export function getCascaderInnerClasses(prefix: string, CLASSNAMES: any, cascaderContext: CascaderContextType) {
  const { disabled, visible, size, multiple } = cascaderContext;
  return [
    `${prefix}-cascader`,
    {
      [CLASSNAMES.STATUS.disabled]: disabled,
      [CLASSNAMES.STATUS.active]: visible,
      [CLASSNAMES.SIZE[size]]: size,
      [`${name}-is-multiple`]: multiple,
    },
  ];
}

/**
 * closeIcon 是否显示
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getCloseShow(isHover: boolean, cascaderContext: CascaderContextType) {
  const { multiple, model, disabled, clearable, visible } = cascaderContext;
  return !!(
    !visible &&
    clearable &&
    isHover &&
    !disabled &&
    ((!multiple && model) || (multiple && (model as TreeNodeValue[]).length))
  );
}

/**
 * Placeholder 是否显示
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getPlaceholderShow(
  cascaderContext: CascaderContextType,
  singleContent: string,
  multipleContent: TreeNode[],
) {
  const { filterable, visible, multiple } = cascaderContext;
  const valEmpty = !(multiple ? multipleContent.length : singleContent);
  if (filterable) {
    return valEmpty && !visible;
  }
  return valEmpty;
}

/**
 * 单选状态下内容
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getSingleContent(cascaderContext: CascaderContextType) {
  const { model, multiple, treeStore, showAllLevels } = cascaderContext;
  if (multiple || !model) return;
  const node = treeStore && treeStore.getNodes(model as TreeNodeValue | TreeNode);
  if (!(node && node.length)) {
    return '';
  }
  const path = node && node[0].getPath();
  if (path && path.length) {
    return showAllLevels ? path.map((node: TreeNode) => node.label).join('/') : path[path.length - 1].label;
  }
  return model as string;
}

/**
 * 多选状态下选中内容
 * @param isHover
 * @param cascaderContext
 * @returns
 */
export function getMultipleContent(cascaderContext: CascaderContextType) {
  const { model, multiple, treeStore } = cascaderContext;

  if (!multiple || !model) return [];
  const node = treeStore && treeStore.getNodes(model as TreeNodeValue | TreeNode);
  if (!node) return [];
  const path = (model as TreeNodeValue[]).map((item: TreeNodeValue) => {
    const node = treeStore.getNodes(item);
    return node[0];
  });
  if (model && (model as TreeNodeValue[]).length) {
    return path && path.length ? path : (model as TreeNode[]);
  }
  return [];
}

/**
 * 点击popup outside副作用
 * @param ref
 * @param cascaderContext
 * @param event
 * @returns
 */
export function outerClickListenerEffect(
  ref: HTMLElement,
  cascaderContext: CascaderContextType,
  event: MouseEvent | TouchEvent,
) {
  const { visible, setVisible } = cascaderContext;
  if (!ref || ref.contains(event.target as Node)) {
    return;
  }

  if (visible) {
    setVisible(false);
  }
}

/**
 * closeIcon点击副作用
 * @param cascaderContext
 */
export function closeIconClickEffect(cascaderContext: CascaderContextType) {
  const { setVisible, setModel, treeStore, multiple } = cascaderContext;

  setModel(multiple ? [] : '');
  treeStore.resetChecked();
  treeStore.resetExpanded();
  setVisible(false);
}

/**
 * tag 关闭按钮点击副作用
 * @param cascaderContext
 */
export function handleRemoveTagEffect(cascaderContext: CascaderContextType, node: TreeNode) {
  const { disabled, setModel } = cascaderContext;

  if (disabled) return;
  const checked = node.setChecked(!node.isChecked());
  setModel(checked);
}

/**
 * 外显content点击副作用
 * @param cascaderContext
 */
export function innerContentClickEffect(cascaderContext: CascaderContextType) {
  const { setVisible, visible, filterActive, disabled } = cascaderContext;

  if (!disabled) {
    if (visible && filterActive) return; // input filter is active
    setVisible(!visible);
  }
}
