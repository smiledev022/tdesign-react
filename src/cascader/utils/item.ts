import { CascaderContextType, TreeNode } from '../interface';

/**
 * 是否显示省略计算方法
 * @param node
 * @param size
 * @returns
 */
export function getLabelIsEllipsis(node: TreeNode, size: CascaderContextType['size']) {
  const sizeMap = {
    small: 11,
    medium: 9,
    large: 8,
  };
  return sizeMap[size] < node.label.length;
}

export function getCascaderItemClass(
  prefix: string,
  node: TreeNode,
  CLASSNAMES: any,
  cascaderContext: CascaderContextType,
) {
  const { checkStrictly, multiple, size } = cascaderContext;
  const expandedActive = !checkStrictly && node.expanded && (multiple ? !node.isLeaf() : true);

  return [
    `${prefix}-cascader-item`,
    {
      [CLASSNAMES.STATUS.selected]: node.checked,
      [CLASSNAMES.STATUS.expanded]: expandedActive,
      [CLASSNAMES.STATUS.disabled]: node.disabled,
      [CLASSNAMES.STATUS.active]: node.actived || (multiple && node.expanded),
      [CLASSNAMES.SIZE[size]]: size,
      [`${prefix}-cascader-item-have-icon`]: node.children,
    },
  ];
}

export function getCascaderItemIconClass(
  prefix: string,
  node: TreeNode,
  CLASSNAMES: any,
  cascaderContext: CascaderContextType,
) {
  const { checkStrictly } = cascaderContext;

  return [
    `${prefix}-cascader-icon`,
    {
      [CLASSNAMES.STATUS.expanded]: !checkStrictly && node.expanded,
    },
  ];
}
