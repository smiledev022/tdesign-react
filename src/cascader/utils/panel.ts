import isFunction from 'lodash/isFunction';
import isNumber from 'lodash/isNumber';
import { TreeNode, CascaderContextType, CascaderProps, ContextType } from '../interface';

/**
 * 面板数据计算方法
 * @param treeNodes
 * @returns
 */
export function getPanels(treeNodes: CascaderContextType['treeNodes']) {
  const panels: TreeNode[][] = [];
  treeNodes.forEach((node: TreeNode) => {
    if (panels[node.level]) {
      panels[node.level].push(node);
    } else {
      panels[node.level] = [node];
    }
  });
  return panels;
}

/**
 * 点击item的副作用
 * @param propsTrigger
 * @param trigger
 * @param node
 * @param cascaderContext
 * @param onChange
 * @param ctx
 */
export function expendClickEffect(
  propsTrigger: CascaderProps['trigger'],
  trigger: CascaderProps['trigger'],
  node: TreeNode,
  cascaderContext: CascaderContextType,
  onChange: CascaderProps['onChange'],
  ctx: any,
) {
  const {
    checkStrictly,
    filterActive,
    multiple,
    treeStore,
    setFilterActive,
    setVisible,
    setModel,
    setTreeNodes,
    setExpend,
  } = cascaderContext;

  // 点击展开节点，设置展开状态
  if (propsTrigger === trigger && !node.isLeaf()) {
    const expanded = node.setExpanded(true);
    treeStore.refreshNodes();
    treeStore.replaceExpanded(expanded);
    const nodes = treeStore.getNodes().filter((node: TreeNode) => node.visible);
    setTreeNodes(nodes);

    // 多选条件下手动维护expend
    if (multiple) {
      setExpend(expanded);
    }
  }

  if (!multiple && (node.isLeaf() || checkStrictly) && trigger === 'click') {
    treeStore.resetChecked();
    const checked = node.setChecked(!node.isChecked());
    const [value] = checked;

    if (onChange && isFunction(onChange)) {
      onChange(value, ctx);
    }

    // 过滤状态下，点击后清除过滤状态
    if (filterActive) {
      setFilterActive(false);
    }

    // 非过滤状态下，关闭
    if (!filterActive) {
      setVisible(false);
    }

    // 非受控状态下更新状态
    setModel(value);
  }
}

/**
 * 多选状态下选中状态数据变化的副作用
 * @param node
 * @param cascaderContext
 * @param onChange
 * @param ctx
 * @returns
 */
export function valueChangeEffect(
  node: TreeNode,
  cascaderContext: CascaderContextType,
  onChange: CascaderProps['onChange'],
  ctx: ContextType,
) {
  const { disabled, max, multiple, setVisible, setModel, filterActive, setFilterActive, treeNodes, treeStore } =
    cascaderContext;

  if (!node || disabled || node.disabled) {
    return;
  }
  const checked = node.setChecked(!node.isChecked());

  if (isNumber(max) && max < 0) {
    console.warn('max should > 0');
  }

  if (checked.length > max && isNumber(max) && max > 0) {
    return;
  }

  if (checked.length === 0) {
    const expanded = treeStore.getExpanded();
    setTimeout(() => {
      treeStore.replaceExpanded(expanded);
      treeStore.refreshNodes();
    }, 0);
  }

  if (!multiple) {
    setVisible(false);
  }

  const isSelectAll = treeNodes.every((item) => checked.indexOf(item.value) > -1);

  if (filterActive && isSelectAll) {
    setVisible(false);
    setFilterActive(false);
  }

  setModel(checked);

  if (onChange && isFunction(onChange)) {
    onChange(checked, ctx);
  }
}
