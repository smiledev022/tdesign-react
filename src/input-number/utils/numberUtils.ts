export const isInValidNumber = (number: number | string) => {
  if (typeof number === 'number') {
    return Number.isNaN(number);
  }

  if (!number) {
    return true;
  }

  return [
    // 普通数字 1.1 1
    /^\s*-?\d+(\.\d+)?\s*$/,
    // 小数点在后 1.
    /^\s*-?\d+\.\s*$/,
    // 小数点在后 .1
    /^\s*-?\.\d+\s*$/,
  ].every((reg) => reg.test(number));
};

export const getNumberPrecision = (number: number | string) => {
  const numStr = String(number);

  return numStr.includes('.') ? numStr.length - numStr.indexOf('.') - 1 : 0;
};

const multiE = (s: string) => {
  const m = s.match(/[e]/gi);
  return m === null ? false : m.length > 1;
};
const multiDot = (s: string) => {
  const m = s.match(/[.]/g);
  return m === null ? false : m.length > 1;
};
const multiNegative = (s: string) => {
  const m = s.match(/[-]/g);
  return m === null ? false : m.length > 1;
};
export const filterInput = (s: string): string => {
  let filterVal = s.replace(/[^\d.eE。-]/g, '').replace('。', '.');
  if (multiE(filterVal) || multiDot(filterVal) || multiNegative(filterVal)) {
    filterVal = filterVal.substr(0, filterVal.length - 1);
  }
  return filterVal;
};
