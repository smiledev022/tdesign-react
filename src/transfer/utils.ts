import isEmpty from 'lodash/isEmpty';
import isFunction from 'lodash/isFunction';
import isString from 'lodash/isString';
import React from 'react';
import { TNode } from '../common';

export const getDefaultValue = <T>(value: T | T[], defaultValue?: T): T[] => {
  if (isEmpty(value)) {
    return Array.isArray(defaultValue) ? defaultValue : [defaultValue, defaultValue];
  }

  if (Array.isArray(value)) return value;

  return [defaultValue, defaultValue];
};

export const getJSX = (value: string | TNode) => {
  if (isString(value)) return value;

  if (isFunction(value)) return value();

  if (React.isValidElement(value)) return value;

  return value;
};
