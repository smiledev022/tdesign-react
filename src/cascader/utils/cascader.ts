import { getFullPathLabel } from './helper';
import { TreeNode, TreeNodeValue, TreeOptionData, CascaderContextType } from '../interface';

/**
 * treeValue计算方法
 * @param model
 * @returns
 */
export const getTreeValue = (model: CascaderContextType['model']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(model)) {
    if (model.length > 0 && typeof model[0] === 'object') {
      treeValue = (model as TreeOptionData[]).map((val) => val.value);
    } else if (model.length) {
      treeValue = model as TreeNodeValue[];
    }
  } else if (model) {
    if (typeof model === 'object') {
      treeValue = [(model as TreeOptionData).value];
    } else {
      treeValue = [model];
    }
  }
  return treeValue;
};

/**
 * input和treeStore变化的副作用
 * @param inputVal
 * @param treeStore
 * @param setTreeNodes
 * @returns
 */
export const treeNodesEffect = (
  inputVal: CascaderContextType['inputVal'],
  treeStore: CascaderContextType['treeStore'],
  setTreeNodes: CascaderContextType['setTreeNodes'],
) => {
  if (!treeStore) return;
  let nodes = [];
  if (inputVal) {
    nodes = treeStore.nodes.filter((node: TreeNode) => {
      const fullPathLabel = getFullPathLabel(node);
      return fullPathLabel.indexOf(inputVal) > -1 && node.isLeaf();
    });
  } else {
    nodes = treeStore.getNodes().filter((node: TreeNode) => node.visible);
  }
  setTreeNodes(nodes);
};

/**
 * 初始化展开阶段与展开状态副作用
 * @param treeStore
 * @param treeValue
 * @param expend
 */
export const treeStoreExpendEffect = (
  treeStore: CascaderContextType['treeStore'],
  treeValue: TreeNodeValue[],
  expend: TreeNodeValue[],
) => {
  // init expanded, 无expend状态时设置
  if (Array.isArray(treeValue) && expend.length === 0) {
    const expandedMap = new Map();
    const [val] = treeValue;
    if (val) {
      expandedMap.set(val, true);
      const node = treeStore.getNode(val);
      if (!node) return;
      node.getParents().forEach((tn: TreeNode) => {
        expandedMap.set(tn.value, true);
      });
      const expandedArr = Array.from(expandedMap.keys());
      treeStore.replaceExpanded(expandedArr);
    }
  }
  // 本地维护 expend，更加可控，不需要依赖于 tree 的状态
  if (treeStore.getExpanded() && expend.length) {
    treeStore.replaceExpanded(expend);
  }
  treeStore.refreshNodes();
};
