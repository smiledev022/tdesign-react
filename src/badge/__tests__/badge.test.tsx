import React from 'react';
import { testExamples, render } from '@test/utils';
import Badge from '../Badge';

// 测试组件代码 Example 快照
testExamples(__dirname);

describe('Badge 组件测试', () => {
  function renderBadge(badge) {
    const { container } = render(badge);
    return container.firstChild;
  }

  test('默认 DOM 结构', async () => {
    const badge = renderBadge(<Badge content={1}>test</Badge>);
    expect(badge.firstChild.textContent).toBe('test');
    expect(badge.lastChild.textContent).toBe('1');
    expect(badge.lastChild).toHaveClass('t-badge--circle');
    expect(badge.childNodes.length).toBe(2);
  });

  test('class', async () => {
    expect(renderBadge(<Badge content={1}>test</Badge>).lastChild).toHaveClass('t-badge--circle');
    expect(
      renderBadge(
        <Badge shape="round" content={1}>
          test
        </Badge>,
      ).lastChild,
    ).toHaveClass('t-badge--round');
    expect(
      renderBadge(
        <Badge dot content={1}>
          test
        </Badge>,
      ).lastChild,
    ).toHaveClass('t-badge--dot');
    expect(
      renderBadge(
        <Badge size="small" content={1}>
          test
        </Badge>,
      ).lastChild,
    ).toHaveClass('t-badge--circle', 't-size-s');

    // not a wrapper
    expect(renderBadge(<Badge content={1} />)).toHaveClass('t-badge-static', 't-badge--circle');
  });

  test('content', async () => {
    // count 0
    expect(renderBadge(<Badge />)).toBeNull();
    expect(renderBadge(<Badge showZero content={0} />)).toBeTruthy();

    // maxCount
    expect(renderBadge(<Badge content={100} />)).toHaveTextContent('99+');
    expect(renderBadge(<Badge content={100} maxCount={100} />)).toHaveTextContent('100');

    // content vs count
    expect(renderBadge(<Badge content={2} />)).toHaveTextContent('2');
  });

  test('offset', async () => {
    expect(renderBadge(<Badge offset={[1, 2]} content={1} />)).toHaveStyle({ right: '-1px', marginTop: '-2px' });
  });

  test('color', async () => {
    expect(renderBadge(<Badge color="red" content={1} />)).toHaveStyle({ backgroundColor: 'red' });
  });
});
