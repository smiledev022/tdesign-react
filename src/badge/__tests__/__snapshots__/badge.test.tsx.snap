// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeColor.jsx 1`] = `
<DocumentFragment>
  <label
    class="t-checkbox t-is-checked"
  >
    <input
      checked=""
      class="t-checkbox__former"
      readonly=""
      type="checkbox"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      红点样式
    </span>
  </label>
  <div>
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(244, 67, 54);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(233, 30, 99);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(156, 39, 176);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(63, 81, 181);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(33, 150, 243);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(3, 169, 244);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(0, 188, 212);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(0, 150, 136);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(76, 175, 80);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(139, 195, 74);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(205, 220, 57);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(255, 235, 59);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(255, 193, 7);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(255, 152, 0);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(255, 87, 34);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(121, 85, 72);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(158, 158, 158);"
    />
    <span
      class="t-badge-static t-badge--dot"
      style="background-color: rgb(96, 125, 139);"
    />
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base t-size-l"
    >
       
    </button>
    <span
      class="t-badge--dot"
    />
  </span>
  <span
    class="t-badge"
  >
    解锁新徽章
    <span
      class="t-badge--dot"
    />
  </span>
  <span
    class="t-badge"
  >
    <svg
      class="t-icon t-icon-user"
      style="font-size: 24px;"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 576C601.92257344 576 686.4307968 599.18159296 759.8745728 639.8946848L760.64 640 797.2160704 658.3197248C818.5708096 669.2764608 832 691.2604224 832 715.261984L832 832C832 867.3462208 803.3462208 896 768 896L256 896C220.653776 896 192 867.3462208 192 832L192 715.2818624C192 691.2790592 205.43055936 669.2941888 226.7869856 658.3380288L263.744 640 264.064 639.936 273.52287808 634.81298816C344.74231744 597.2582336 425.8885952 576 512 576ZM512 640C416.8213984 640 328.5751328 669.6808576 256.01207488 720.2916928L256 832 768 832 768.0210368 720.3147904C695.4517184 669.689888 607.19308032 640 512 640ZM512 159.72895488C618.038672 159.72895488 704 245.69028288 704 351.72895488 704 457.76762624 618.038672 543.72895488 512 543.72895488 405.961328 543.72895488 320 457.76762624 320 351.72895488 320 245.69028288 405.961328 159.72895488 512 159.72895488ZM512 223.72895488C441.307552 223.72895488 384 281.03650688 384 351.72895488 384 422.42140224 441.307552 479.72895488 512 479.72895488 582.692448 479.72895488 640 422.42140224 640 351.72895488 640 281.03650688 582.692448 223.72895488 512 223.72895488Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class="t-badge--dot"
    />
  </span>
</DocumentFragment>
`;

exports[`custom.jsx 1`] = `
<DocumentFragment>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base t-size-l"
    >
       
    </button>
    <span
      class="t-badge--round"
    >
      new
    </span>
  </span>
</DocumentFragment>
`;

exports[`number.jsx 1`] = `
<DocumentFragment>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base t-size-l"
    >
       
    </button>
    <span
      class="t-badge--circle"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base t-size-l"
    >
       
    </button>
    <span
      class="t-badge--circle"
    >
      99
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base t-size-l"
    >
       
    </button>
    <span
      class="t-badge--circle"
    >
      99+
    </span>
  </span>
</DocumentFragment>
`;

exports[`offset.jsx 1`] = `
<DocumentFragment>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      默认
    </button>
    <span
      class="t-badge--circle"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      [10,10]
    </button>
    <span
      class="t-badge--circle"
      style="right: -10px; margin-top: -10px;"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      [-10,10]
    </button>
    <span
      class="t-badge--circle"
      style="right: 10px; margin-top: -10px;"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      [-10,-10]
    </button>
    <span
      class="t-badge--circle"
      style="right: 10px; margin-top: 10px;"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      [10,-10]
    </button>
    <span
      class="t-badge--circle"
      style="right: -10px; margin-top: 10px;"
    >
      2
    </span>
  </span>
</DocumentFragment>
`;

exports[`shape.jsx 1`] = `
<DocumentFragment>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
       circle
    </button>
    <span
      class="t-badge--circle"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      round
    </button>
    <span
      class="t-badge--round"
    >
      99
    </span>
  </span>
</DocumentFragment>
`;

exports[`size.jsx 1`] = `
<DocumentFragment>
  <h3
    style="margin-bottom: 16px;"
  >
    1.默认大小
  </h3>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      按钮
    </button>
    <span
      class="t-badge--circle"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      按钮
    </button>
    <span
      class="t-badge--circle"
    >
      99
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      按钮
    </button>
    <span
      class="t-badge--circle"
    >
      99+
    </span>
  </span>
  <h3
    style="margin-bottom: 16px; margin-top: 16px;"
  >
    2.小
  </h3>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      按钮
    </button>
    <span
      class="t-badge--circle t-size-s"
    >
      2
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      按钮
    </button>
    <span
      class="t-badge--circle t-size-s"
    >
      99
    </span>
  </span>
  <span
    class="t-badge"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      按钮
    </button>
    <span
      class="t-badge--circle t-size-s"
    >
      99+
    </span>
  </span>
</DocumentFragment>
`;
